src/main/java/com/uber/jenkins/phabricator/coverage/JacocoCoverageProvider.java

```diff
@@ -79,26 +76,26 @@ protected CodeCoverageMetrics getCoverageMetrics() {
             return null;
         }
 
-        PathResolver pathResolver = new PathResolver(getWorkspace(), getSourceDirs());
         HashMap<String, List<Integer>> lineCoverage = new HashMap<String, List<Integer>>();
 
         String[] includes = null;
         if (jacocoPublisher.getInclusionPattern() != null) {
-            includes = new String[]{ jacocoPublisher.getInclusionPattern() };
+            includes = new String[] {jacocoPublisher.getInclusionPattern()};
         }
 
         String[] excludes = null;
         if (jacocoPublisher.getExclusionPattern() != null) {
-            excludes = new String[]{jacocoPublisher.getExclusionPattern()};
+            excludes = new String[] {jacocoPublisher.getExclusionPattern()};
         }
 
         try {
             ExecutionFileLoader executionFileLoader = jacocoAction.getJacocoReport().parse(includes, excludes);
             for (IPackageCoverage packageCoverage : executionFileLoader.getBundleCoverage().getPackages()) {
-                for (ISourceFileCoverage fileCoverage : packageCoverage.getSourceFiles()) {
-                    String relativePathFromProjectRoot = getRelativePathFromProjectRoot(pathResolver, fileCoverage);
+                for (IClassCoverage classCoverage : packageCoverage.getClasses()) {
+                    String relativePathFromProjectRoot = getRelativePathFromProjectRoot(
+                            classCoverage.getSourceFileName());
                     if (relativePathFromProjectRoot != null) {
-                        lineCoverage.put(relativePathFromProjectRoot, getPerLineCoverage(fileCoverage));
+                        lineCoverage.put(relativePathFromProjectRoot, getPerLineCoverage(classCoverage));
```

Comment:
Won't it change result behaviour? Looks like you're changing fileCoverage to class coverage? Is this related to performance?

Reply:
Going to revert the behavior back